src/components/JoinExistingList.jsx

```diff
@@ -0,0 +1,53 @@
+import { db } from '../lib/firebase';
+import { collection, onSnapshot } from '@firebase/firestore';
+import { useState, useEffect } from 'react';
+import { setUser } from '../storage-utils/storage-utils';
+import { useNavigate } from 'react-router-dom';
+
+export default function JoinExistingList() {
+  const [userToken, setUserToken] = useState('');
+  const [docs, setDocs] = useState([]);
+  const navigate = useNavigate();
+
+  // get the collection when userToken changes
+  useEffect(() => {
+    if (userToken !== '') {
+      const unsubscribe = onSnapshot(collection(db, userToken), (snapshot) => {
+        let snapshotDocs = [];
+        snapshot.forEach((doc) => snapshotDocs.push(doc));
+        setDocs(snapshotDocs);
+      });
+      return () => {
+        unsubscribe();
+      };
+    }
+  }, [userToken]);
+
+  return (
+    <div>
+      <p>Join an existing list by entering a three word token below.</p>
+
+      <form
+        onSubmit={() => {
+          if (docs.length > 0) {
+            setUser(userToken);
+            navigate('list');
+          } else {
+            alert(
+              'That list does not exist. Please try again or create a new list.',
```

Comment:
As we're not checking the Firestore DB here for existence (we could possibly, in the future), would it make more sense to just use an error along the lines of  "Invalid List ID. Please try again or create a new list."

Reply:
We are checking for the existence of the collection on line 13-24 in the useEffect. That sets the value of `docs` to an array of all of the docs in the collection...unless it doesn't exist. In that case, it returns an empty array 😃

Reply:
Makes sense, thanks for the explanation!